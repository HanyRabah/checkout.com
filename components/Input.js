import styled from 'styled-components'
import Tooltip from './ToolTip'

const FormInput = (props) => {
  const {
    type,
    label,
    name,
    placeholder,
    errorText,
    handleChange,
    value,
    row,
    required,
  } = props

  return (
    <InputWrapper>
      <Label>
        {required && <Required title="required">*</Required>} {label}
      </Label>
      {type === 'text' && (
        <Input
          name={name}
          placeholder={placeholder}
          isInvalid={errorText}
          onChange={handleChange}
          value={value}
        />
      )}
      {type === 'textArea' && (
        <TextArea
          name={name}
          placeholder={placeholder}
          isInvalid={errorText}
          onChange={handleChange}
          value={value}
          rows={row}
        />
      )}
      {errorText && <Tooltip text={errorText} />}
    </InputWrapper>
  )
}

const InputWrapper = styled.div`
  margin-bottom: 10px;
  position: relative;
`

const Label = styled.label`
  font-size: 14px;
  margin-bottom: 8px;
  font-weight: 600;
`

const Input = styled.input`
  display: block;
  background: #ececec;
  border-radius: 4px;
  width: 100%;
  height: 50px;
  border: 0;
  outline: 0;
  font-size: 14px;
  text-indent: 12px;
  line-height: 14;
  ${(props) => props.isInvalid && `border: 1px red solid;`}
  &:focus {
    background: #f1f1f1;
  }
`
const TextArea = styled.textarea`
  display: block;
  background: #ececec;
  border-radius: 4px;
  width: 100%;
  border: 0;
  outline: 0;
  font-size: 14px;
  text-indent: 12px;
  line-height: 2;

  ${(props) => props.isInvalid && `border: 1px red solid;`}

  &:focus {
    background: #f1f1f1;
  }
`
const Rating = styled.div`
  display: flex;
`
const Required = styled.span`
  color: #fd496d;
`

export default FormInput
