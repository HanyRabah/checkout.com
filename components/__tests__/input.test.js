import React from 'react'
import Input from '../Input'
import renderer from 'react-test-renderer'
import { cleanup, fireEvent, render } from '@testing-library/react'

const setupTextInput = () => {
  const { getByPlaceholderText } = render(
    <Input
      type="text"
      placeholder="name"
      label="What is your name ?"
      name="name"
    />,
  )
  const input = getByPlaceholderText('name')
  return { input }
}

const setupTextArea = () => {
  const { getByPlaceholderText } = render(
    <Input
      type="textArea"
      placeholder="Comments"
      label="Leave your comment"
      name="comment"
    />,
  )
  const textArea = getByPlaceholderText('Comments')
  return { textArea }
}

describe('Input', () => {
  afterEach(cleanup)

  it('render correctly', () => {
    const component = renderer.create(
      <Input
        type="text"
        placeholder="Name"
        label="What is your name ?"
        name="name"
      />,
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('It should change value', () => {
    const { input } = setupTextInput()
    expect(input.value).toBe('') // empty before
    fireEvent.change(input, { target: { value: 'Good Day' } })
    expect(input.value).toBe('Good Day')
  })
})

describe('TextArea', () => {
  afterEach(cleanup)

  it('render correctly', () => {
    const component = renderer.create(
      <Input
        type="textArea"
        placeholder="Comments"
        label="Leave your comment"
        name="comment"
      />,
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('It should change value', () => {
    const { textArea } = setupTextArea()
    expect(textArea.value).toBe('') // empty before
    fireEvent.change(textArea, { target: { value: 'Here is my comment' } })
    expect(textArea.value).toBe('Here is my comment')
  })
})
