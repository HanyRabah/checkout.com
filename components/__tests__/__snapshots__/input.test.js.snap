// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input render correctly 1`] = `
<div
  className="Input__InputWrapper-p4pzeo-0 iLsCTM"
>
  <label
    className="Input__Label-p4pzeo-1 inQhSJ"
  >
     
    What is your name ?
  </label>
  <input
    className="Input-p4pzeo-2 kEMRak"
    name="name"
    placeholder="Name"
  />
</div>
`;

exports[`TextArea render correctly 1`] = `
<div
  className="Input__InputWrapper-p4pzeo-0 iLsCTM"
>
  <label
    className="Input__Label-p4pzeo-1 inQhSJ"
  >
     
    Leave your comment
  </label>
  <textarea
    className="Input__TextArea-p4pzeo-3 jXbrWD"
    name="comment"
    placeholder="Comments"
  />
</div>
`;
